pkg/controllers/raycluster_webhook.go

```diff
@@ -0,0 +1,158 @@
+	raycluster := obj.(*rayv1.RayCluster)
+
+	rayclusterlog.Info("default", "name", raycluster.Name)
+	// Check and add OAuth proxy if it does not exist.
+	alreadyExists := false
+	for _, container := range raycluster.Spec.HeadGroupSpec.Template.Spec.Containers {
+		if container.Name == "oauth-proxy" {
+			rayclusterlog.Info("OAuth sidecar already exists, no patch needed")
+			alreadyExists = true
+			break // exits the for loop
+		}
+	}
+
+	if !alreadyExists {
+		rayclusterlog.Info("Adding OAuth sidecar container")
+		// definition of the new container
+		newOAuthSidecar := corev1.Container{
+			Name:  "oauth-proxy",
+			Image: "registry.redhat.io/openshift4/ose-oauth-proxy@sha256:1ea6a01bf3e63cdcf125c6064cbd4a4a270deaf0f157b3eabb78f60556840366",
+			Ports: []corev1.ContainerPort{
+				{ContainerPort: 8443, Name: "oauth-proxy"},
+			},
+			Args: []string{
+				"--https-address=:8443",
+				"--provider=openshift",
+				"--openshift-service-account=" + raycluster.Name + "-oauth-proxy",
+				"--upstream=http://localhost:8265",
+				"--tls-cert=/etc/tls/private/tls.crt",
+				"--tls-key=/etc/tls/private/tls.key",
+				"--cookie-secret=$(COOKIE_SECRET)",
+				"--openshift-delegate-urls={\"/\":{\"resource\":\"pods\",\"namespace\":\"default\",\"verb\":\"get\"}}",
+				"--added-label=True",
+			},
+			Env: []corev1.EnvVar{
+				{
+					Name: "COOKIE_SECRET",
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: raycluster.Name + "-oauth-config",
+							},
+							Key: "cookie_secret",
+						},
+					},
+				},
+			},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      "proxy-tls-secret",
+					MountPath: "/etc/tls/private",
+					ReadOnly:  true,
+				},
+			},
+		}
+
+		// Adding the new OAuth sidecar container
+		raycluster.Spec.HeadGroupSpec.Template.Spec.Containers = append(raycluster.Spec.HeadGroupSpec.Template.Spec.Containers, newOAuthSidecar)
+
+		tlsSecretVolume := corev1.Volume{
+			Name: "proxy-tls-secret",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: raycluster.Name + "-proxy-tls-secret",
+				},
+			},
+		}
+
+		raycluster.Spec.HeadGroupSpec.Template.Spec.Volumes = append(raycluster.Spec.HeadGroupSpec.Template.Spec.Volumes, tlsSecretVolume)
+
+		// Ensure the service account is set
+		if raycluster.Spec.HeadGroupSpec.Template.Spec.ServiceAccountName == "" {
+			raycluster.Spec.HeadGroupSpec.Template.Spec.ServiceAccountName = raycluster.Name + "-oauth-proxy"
+		}
+	}
+	return nil
+}
+
+func (v *RayClusterValidator) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	raycluster := obj.(*rayv1.RayCluster)
+	var warnings admission.Warnings
+	var allErrors field.ErrorList
+	specPath := field.NewPath("spec")
+
+	if raycluster.Spec.HeadGroupSpec.EnableIngress == nil || *raycluster.Spec.HeadGroupSpec.EnableIngress {
```

Comment:
The default behavior for EnableIngress is equivalent to EnableIngress=False afaict. Can we allow users to just not specify it?

https://github.com/ray-project/kuberay/blob/791ea372a72900f47ca2e1166751c5219c83d8c0/ray-operator/controllers/ray/raycluster_controller.go#L425

Reply:
I agree, unspecified `enableIngress` should be treated as disabled, and be allowed.

Reply:
Good stuff, thank you guys!